### ISSUE TITLE
Delete logs from watch

### ISSUE THREAD

**Issue by vascotenner** [openambitproject/openambit#213]
Can openambit/libopenambit be used to delete moves/log from the watch? How?

**Comment by Gerold-R** [comment]
As far as I know this is not possible. 

You can use the function "Force Firmware Update" from Moveslink to delete the complete logbook. After the update, the header of the ringbuffer for the exercise log is reset to it's default values. It's possible to write a small tool that can delete the complete logook by overwritting these 18 header bytes with it's default values. For the Ambit 2 series the header is stored at address $0F4240, for the Ambit 3 series it's $0027AC40.

I only deleted the logbook of an Ambit 2S once and it worked.

**Comment by vascotenner** [comment]
For me, the reason to use openambit is two-fold:
1. Runs on linux
2. Synchronization of the watch without uploading the logs to suuntos webserver

@Gerold-R : Do you have sample code how to do this?

**Comment by markusdecker** [comment]
hi vasco

> For me, the reason to use openambit is two-fold:
> 1. Runs on linux
> 2. Synchronization of the watch without uploading the logs to suuntos webserver

you are able to sync the sportmodes with your watch?

m 


> @Gerold-R : Do you have sample code how to do this?
> 
> -- 
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/openambitproject/openambit/issues/213#issuecomment-452267328

-- 
firstfloor.org/ae

**Comment by Gerold-R** [comment]
Yes of course, I do have code but it's written in Delphi 7. Do you have an Ambit 2S or another model? 

Where do you upload your moves if you dont use Suunto's WebServer?

BTW: 
I also using Linux (Mint) and I am a CrossOver Advocate for Moveslink
https://www.codeweavers.com/compatibility/crossover/moveslink

Mainly I am using programs which runs on Linux, only for a new windows programs (and my own ones) I use CrossOver (before I used WINE but I prefer CrossOver now). I was able to run original Moveslink on Linux using WINE or CrossOver (latest version 17). It was always a challenge to get Moveslink to run on Linux again after a new version of WINE or CrossOver, so I have written new a GUI using original DLL from Moveslink to sync the watch. It's like a workaround until I find a better solution and described here:
https://www.codeweavers.com/compatibility/crossover/tips/moveslink

Note, that you need Movescount.com to setup your watch, even if you don't want to upload the logs.

**Comment by markusdecker** [comment]
hi

i am useing debian. and i have the ambit peak 3. i didnt figure out how to program the watch
on linux. if theres a way it would be much appreciated, but as far as i understand ambit 3 is not supported.

ciao
m

**Comment by rnorris** [comment]
My Ambit 3 Multisport works with Debian using the latest source code.

I haven't knowingly used movescount.com (unless maybe it's used for 'Orbit Data' whatever that is)

**Comment by markusdecker** [comment]
hi rnorris.
interessting, and you are able to sync your sportmodes? did i understand this right?
how you do this?
m

**Comment by rnorris** [comment]
> hi rnorris.
> interessting, and you are able to sync your sportmodes? did i understand this right?
> how you do this?
> m

I just use openambit to download the logs to my computer (which I then convert to GPX).

**Comment by fmoerl-real** [comment]
Now, the movescount service is closed. Is there any linux-based possibility to delete the logs from the watch. I ask this, because my Ambit 2 prompts the "Sync to movescount" screen due to more than 50% usage of memory. Is this only a flag? If so, can I ignore it or will my watch actually run in a state of out of memory?

**Comment by rnorris** [comment]
@fmoerl-real: I believe the Ambit will automatically deletes/overwrites the oldest logs when it becomes full, so you should always have the latest logs available. I've not synced/manually deleted anything on my (Ambit3) watch for 2 years or so.

**Comment by centic9** [comment]
You can run the sync via the commandline-tool openambit-cli with some options to disable syncing to movescount while still downloading moves and marking them as "fetched" on the watch, thus removing that warning.

Also some have reported success in still being able to upload to movescount with new credentials, see https://github.com/openambitproject/openambit/issues/270#issuecomment-782908757 for some details.

**Comment by fmoerl-real** [comment]
Thank you, centic9, for the hint. I figured out [openambit-cli -tosn] would read my data without to try syncing it to movescount. If it really works, I will find out after my next activity. Is there anywhere a more detailed description how to use the options, for instance the -c option is not clear without an example file. 

Movescount is not important for me. I can read my logs, translate it a *.gpx file and then I do a manual upload to strava.
